main/src/sphexa/propagator.hpp

```diff
@@ -270,6 +301,18 @@ class HydroVeProp final : public Propagator<DomainType, ParticleDataType>
         timer.step("mpi::synchronizeHalos");
         computeGradPVE(first, last, ngmax_, d, domain.box());
         timer.step("MomentumAndEnergy");
+
+        d.egrav = 0.0;
+        if (doGravity_)
+        {
+            mHolder_.upsweep(d, domain);
+            timer.step("Upsweep");
+            mHolder_.traverse(d, domain);
+            // temporary sign fix, see note in ParticlesData
+            d.egrav = (d.g > 0.0) ? d.egrav : -d.egrav;
```

Comment:
Why do we need this check here? G is always positive. Either 1 or 6.67e-8.

Reply:
Yes indeed. The gravity module reports acceleration, whereas momentum_energy reports negative acceleration.
This negative-G hack is used to reconcile that discrepancy. IMHO, the simplest solution would be to make momentum_energy j-loop return grad_P by value (instead of direct memory access with a pointer), such that we can keep the usual convention of returning gradient of pressure for the j-loop function. The calling function, that does the i-loop can then decide to store grad_P back to memory as acceleration with the negative sign.